CHANGELOG.md

```diff
@@ -99,6 +79,254 @@ importing them.
+
+- Fills can now be defined within loops (`{% for %}`) or other tags (like `{% with %}`),
+  or even other templates using `{% include %}`.
+  
+  Following is now possible
+
+  ```django
+  {% component "table" %}
+    {% for slot_name in slots %}
+      {% fill name=slot_name %}
+      {% endfill %}
+    {% endfor %}
+  {% endcomponent %}
+  ```
+
+- If you need to access the data or the default content of a default fill, you can omit
+  the `name` kwarg, or set it to `"default"`.
+
+  Previously, a default fill would be defined simply by omitting the `{% fill %}` tags:
+
+  ```django
+  {% component "child" %}
+    Hello world
+  {% endcomponent %}
+  ```
+
+  But in that case you could not access the slot data or the default content, like it's possible
+  for named fills:
+  
+  ```django
+  {% component "child" %}
+    {% fill name="header" data="data" %}
+      Hello {{ data.user.name }}
+    {% endfill %}
+  {% endcomponent %}
+  ```
+
+  Now, you can specify default tag by either omitting the `name` kwarg, or using `name="default"`:
+
+  ```django
+  {% component "child" %}
+    {% fill data="data" %}
+      Hello {{ data.user.name }}
+    {% endfill %}
+  {% endcomponent %}
+
+  {% component "child" %}
+    {% fill name="default" data="data" %}
+      Hello {{ data.user.name }}
+    {% endfill %}
+  {% endcomponent %}
+  ```
+
+- When inside `get_context_data()` or other component methods, the default fill
+  can now be accessed as `Component.input.slots["default"]`, e.g.:
+
+  ```py
+  class MyTable(Component):
+      def get_context_data(self, *args, **kwargs):
+          default_slot = self.input.slots["default"]
+          ...
+  ```
+
+- You can now dynamically pass all slots to a child component. This is similar to
+  [passing all slots in Vue](https://vue-land.github.io/faq/forwarding-slots#passing-all-slots):
+
+  ```py
+  class MyTable(Component):
+      def get_context_data(self, *args, **kwargs):
+          return {
+              "slots": self.input.slots,
+          }
+
+      template: """
+        <div>
+          {% component "child" %}
+            {% for slot_name in slots %}
+              {% fill name=slot_name data="data" %}
+                {% slot name=slot_name ...data / %}
+              {% endfill %}
+            {% endfor %}
+          {% endcomponent %}
+        </div>
+      """
+  ```
+
+#### Fix
+
+- Slots defined with `{% fill %}` tags are now properly accessible via `self.input.slots` in `get_context_data()`
+
+- Do not raise error if multiple slots with same name are flagged as default
+
+- Slots can now be defined within loops (`{% for %}`) or other tags (like `{% with %}`),
+  or even other templates using `{% include %}`.
+  
+  Previously, following would cause the kwarg `name` to be an empty string:
+
+  ```django
+  {% for slot_name in slots %}
+    {% slot name=slot_name %}
+  {% endfor %}
+  ```
+
+#### Refactor
+
+- When you define multiple slots with the same name inside a template,
```

Comment:
As I was going through the README, I realized there's one more change of behavior - when we have multiple `{% slot %}` tags with the same name, the `default` and `required` flag are now applied only to the individual tags